sharktank/sharktank/types/tensors.py

```diff
@@ -752,10 +752,29 @@ class ShardedTensor(InferenceTensor):
     """
 
     def __init__(
-        self, *, shape: list[int], shard_dim: int | None, name: str = UnnamedTensorName
+        self, *, 
+        shape: list[int], 
+        shard_dim: int | None, 
+        name: str = UnnamedTensorName,
+        devices: Tuple[int],
+        devices_pinned: bool,
```

Comment:
Make this an `Optional[bool]` set to  `None`. From there we can check some behaviors. We may be able to check if `torch.Tensor` is a `AbstractTensor` or one associated with the value to pin.

Overall we want to think of this default as "if this is real values, has a DeviceTensorTrait, etc lets make it pinned".

Reply:
See discussion on your other point: https://github.com/nod-ai/shark-ai/pull/1008#discussion_r1972422563